service/matching/handler/engine_integration_test.go

```diff
@@ -1527,8 +1533,14 @@ func addTask(engine *matchingEngineImpl, hCtx *handlerContext, request *addTaskR
 			ForwardedFrom:                 request.ForwardedFrom,
 			PartitionConfig:               request.PartitionConfig,
 		})
+		if err != nil {
+			return nil, err
+		}
+		return &addTaskResponse{
+			PartitionConfig: resp.PartitionConfig,
+		}, nil
 	}
-	return engine.AddDecisionTask(hCtx, &types.AddDecisionTaskRequest{
+	resp, err := engine.AddDecisionTask(hCtx, &types.AddDecisionTaskRequest{
```

Comment:
why not return the `resp` from this helper function instead of creating a new struct and filling it?

Reply:
Because Activity and Decision have different response types.